content/blog/2021/compact_filters_demo.md

```diff
@@ -0,0 +1,367 @@
+
+In this tutorial, We will use `bdk-cli` to demonstrate some basic BDK functionalities as a Neutrino wallet.
+
+## Tutorial Scope
+To demonstrate basic wallet functionalities, with Neutrino backend using `bdk-cli`
+
+  - create and sync a wallet,
+  - receive a transaction,
+  - create a transaction,
+  - sign and broadcast the transaction,
+  - fetch updated balance,
+
+Here we will have a Bitcoin Core wallet and a BDK wallet, sending and receiving transactions between each other. The BDK wallet will sync itself using compact filters from the Core node. 
+
+## Prerequisites
+Following things are required to start with the tutorial.
+
+1. A Bitcoin Core regtest node listening at `localhost:18444` signalling for compact filter support. 
+2. `bdk-cli` compiled with `compact_filter` features.
+
+If you already have these two setup and working, you can skip this and jump to the [Tutorial](#Tutorial) section.
+
+#### Install and run `bitcoind` 
+You can definitely do it with your own `bitcoind` installation. `BIP157` support has been included in Bitcoin Core `v0.21.0`. So anything above that will work.
+
+You also need to ensure proper configuration settings for signalling `compact_filters` support.  
+
+For ease of testing, BDK project hosts docker images that can be used to spawn Bitcoin Core with all the relevant configurations.
+
+- spawn a regtest node using [bitcoin-regtest-box](https://github.com/bitcoindevkit/bitcoin-regtest-box) docker file. 
+  
+  This should start logging the `debug.log` in the terminal. You can keep this terminal alive to see communication events with BDK and the node.
+
+  ```shell
+  $ docker run -p 127.0.0.1:18443-18444:18443-18444/tcp --name bdk-box bitcoindevkit/bitcoind
+  ```
+  This will spin up a docker container running `bicoind` and listening to port `18444` and `18333`.
+
+  ```shell
+  $ sudo netstat -nptl | grep 'docker'
+    tcp    0   0   127.0.0.1:18443     0.0.0.0:*      LISTEN  140708/docker-proxy
+    tcp    0   0   127.0.0.1:18444     0.0.0.0:*      LISTEN  140695/docker-proxy
+  ```
+- Check node is reachable
+
+  Assuming you have `bitcoin-cli` already installed, 
+  In another terminal try connecting to the node with `bitcoin-cli`
+  ```shell
+  $ bitcoin-cli getnetworkinfo
+  {
+    "version": 210000,
+    "subversion": "/Satoshi:0.21.0/",
+    "protocolversion": 70016,
+    "localservices": "0000000000000449",
+    "localservicesnames": [
+      "NETWORK",
+      "WITNESS",
+      "COMPACT_FILTERS",
+      "NETWORK_LIMITED"
+      ...
+    ],
+  }
+
+  ```
+  The `version` byte should be `210000`. `localservicesnames` should contain `"COMPACT_FILTERS"`. If you see this, then Bitcoin Core is correctly configured.
+
+#### Install and run bdk-cli
+- Install `bdk-cli` with `compact_filters` feature
+
+  ```shell
+  $ cargo install bdk-cli --features compact_filters
```

Comment:
Package `bdk-cli v0.2.0` does not have the feature `compact_filters`

Reply:
Yes, because https://github.com/bitcoindevkit/bdk-cli/pull/26 hasn't been merged yet. You can build `bdk-cli` on top of that PR branch to get `compact_filters` support.

Reply:
oops, sorry!

Reply:
got the correct version from your pr branch now :P